tools/bwa.py

```diff
@@ -46,18 +48,28 @@ def index(self, inFasta, algorithm=None):
         cmd.append(inFasta)
         self.execute('index', cmd)
 
-    def mem(self, inReads, refDb, outAlign, threads=None):
+    def mem(self, inReads, refDb, outAlign, opts=None, threads=None):
+        opts = [] if not opts else opts
+
         threads = threads or util.misc.available_cpu_count()
         samtools = tools.samtools.SamtoolsTool()
         fq1 = util.file.mkstempfname('.1.fastq')
         fq2 = util.file.mkstempfname('.2.fastq')
         aln_sam = util.file.mkstempfname('.sam')
+        aln_sam_sorted = util.file.mkstempfname('sorted.sam')
         samtools.bam2fq(inReads, fq1, fq2)
-        self.execute('mem', ['-t', str(threads), refDb, fq1, fq2], stdout=aln_sam)
+        self.execute('mem', opts + ['-t', str(threads), refDb, fq1, fq2], stdout=aln_sam)
```

Comment:
may need to pull the threads param out of the self.execute array here and do something like:
```
if '-t' not in opts:
    threads = threads or utils.misc.available_cpu_count()
    opts.extend(('-t', str(threads))
```
then just `self.execute('mem', opts + [refDb, fq1, fq2], stdout=aln_sam)`